geth/jail/jail_cell_test.go

```diff
@@ -109,3 +107,190 @@ func (s *JailTestSuite) TestJailLoopInCall() {
 		require.Fail("Failed to received event response")
 	}
 }
+
+func (s *JailTestSuite) TestJailLoopRace() {
+	require := s.Require()
+	require.NotNil(s.jail)
+
+	cell, err := s.jail.NewCell(testChatID)
+	require.NoError(err)
+	require.NotNil(cell)
+
+	items := make(chan string)
+
+	err = cell.Set("__captureResponse", func(val string) otto.Value {
+		go func() { items <- val }()
+		return otto.UndefinedValue()
+	})
+	require.NoError(err)
+
+	_, err = cell.Run(`
+		function callRunner(namespace){
+			return setTimeout(function(){
+				__captureResponse(namespace);
+			}, 1000);
+		}
+	`)
+	require.NoError(err)
+
+	for i := 0; i < 100; i++ {
+		_, err = cell.Call("callRunner", nil, "softball")
+		require.NoError(err)
+	}
+
+	for i := 0; i < 100; i++ {
+		select {
+		case received := <-items:
+			require.Equal(received, "softball")
+			break
```

Comment:
`break` in `select`? After `require`? You seem to really treasure safety :)

Reply:
Oops, nice catch :)